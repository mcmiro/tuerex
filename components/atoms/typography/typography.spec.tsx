import * as React from "react";
import {screen, render} from "@testing-library/react";
import Typography from "./";

describe("Test ThumbnailItem component", () => {
	it("Should pass when variant is defined", () => {
		render(<Typography variant="md" data-testid="typography-test" />);

		const typography = screen.queryByTestId("typography-test");
		expect(typography).toBeDefined();
		expect(typography).toMatchSnapshot();
	});

	it("Should pass when variant change html to paragraph", () => {
		const {container} = render(<Typography variant="md" data-testid="typography-test" />);
		expect(container.innerHTML).toBe('<p class="m-0 p-0 whitespace-pre-line text-md"></p>');
	});

	it("Should pass when props weight change to bold", () => {
		const {container} = render(<Typography variant="md" weight="semibold" />);
		expect(container.innerHTML).toBe('<p class="m-0 p-0 whitespace-pre-line text-md font-semibold"></p>');
	});
});
