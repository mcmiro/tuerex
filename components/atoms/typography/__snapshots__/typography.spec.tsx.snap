// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ThumbnailItem component Should pass when variant is defined 1`] = `null`;
